src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -116,20 +116,49 @@ private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string
 
         var hasBodyOrFormFileParameter = false;
 
-        foreach (var parameter in methodInfo.GetParameters())
+        static List<ParameterInfo> FlattenParameters(ParameterInfo[] parameters)
         {
-            var parameterDescription = CreateApiParameterDescription(parameter, routeEndpoint.RoutePattern, disableInferredBody);
+            var flattenedParameters = new List<ParameterInfo>();
```

Comment:
Ideally we wouldn't allocate if there was nothing to flatten.

Reply:
Fixed